### DISCUSSION TITLE
Parsing antlers string from one collection with data from another?

### DISCUSSION THREAD

**Discussion by jasonpolito** [statamic/cms#8492] [Q&A] [answered]
Ok, so this may be the completely wrong way to go about / think of things but I'm basically trying to allow a user to manage antlers templates between two separate collections. One collection has a bard field with sets, the other just has "markup".

Say I have two collections:

```
- Pages
  - Bard field with sets

- Views
  - Handle
  - Markup
```

An example set from the `pages` bard field could be something like this:

```
type: hero
title: 'This is the title!'
content: 'Here is some example content.'
```

...and an `hero` example entry from `views`:

```
-
handle: 'hero'
markup:
  <div>
      <h1>{{ title }}</h1>
      <div>{{ content }}</div>
  </div>
```

Is it possible to parse the `hero` markup using the data from the `page` entry?

```antlers
{{?
    use Statamic\Facades\Entry;

    $view = Entry::query()
        ->where('collection', 'views')
        ->where('slug', Str::of($type)->slug('-'))
        ->first();
    
    $markup = $view->get('markup')['code'];
?}}

{{ markup }}
```

As in, `{{ markup }}` from `views` would be parsed by antlers with access to `{{ title }}` and `{{ content }}` from `pages`? Is this possible? The idea is being able to manage the antlers views directly from within Statamic vs managing actual `[name].antlers.html` files from within the codebase.

Thank you for any suggestions!

**Comment by ryanmitchell** [comment]
You should be able to do that using the antlers facade.

```php
Statamic\Facades\Antlers::parse($markup, $data);
```

**Reply by jasonpolito** [reply]
Thank you @ryanmitchell, I'm just not too sure how to get the `page` variables in to `$data` since it wouldn't be the same variables each time. Is there a way to access the current scope?

```antlers
{{ content }}
    {{ partial:render_view }}
{{ /content }}
```

```antlers
{{# render_view.antlers.html #}}
{{?
    use Statamic\Facades\Entry;
    use Statamic\Facades\Antlers;

    $view = Entry::query()
        ->where('collection', 'views')
        ->where('slug', Str::of($type)->slug('-'))
        ->first();

    $raw = $view->get('markup')['code'];
    
    // would like to get variables into $data here
    // is there something like $context that's accessible?

    $markup = Antlers::parse($raw, $data);
?}}

{{ markup }}
```

**Reply by ryanmitchell** [reply]
Not too sure - you can {{ dump }} to see whats available.
Maybe $context ?

**Reply by JohnathonKoster** [reply, marked as answer]
An Antlers-only way might look something like this (with the main addition being the `antlers` modifier):

```antlers
{{
    _entry = ({collection:views slug:is="hero" limit="1"} | first);
    _entry.markup.code | antlers
/}}

```

**Reply by jasonpolito** [reply]
@JohnathonKoster this is absolutely perfect -- exactly what I was trying to do! Thanks so much for your help and the amazing work you do.
